capstone/src/main/java/com/google/step/clients/MetricsClient.java

```diff
@@ -179,7 +173,7 @@ public boolean putPageView(String restaurantKey) {
                 // Query for the name
                 Query nameQuery = new Query("RestaurantInfo")
                         .setFilter(new FilterPredicate(
-                                "restaurantKey", Query.FilterOperator.EQUAL, restaurantKey));
+                                Entity.KEY_RESERVED_PROPERTY, Query.FilterOperator.EQUAL, KeyFactory.createKey("RestaurantInfo", Long.parseLong(restaurantKey))));
```

Comment:
Just to clarify something, so the frontend always has the key of the restaurant its displaying right?